core/postInitClient.sqf

```diff
@@ -29,34 +29,111 @@
+                _last_hit_info params ["_data", "_time", "_unitPos", "_killerPos"];
+                if (_time + 10 > CBA_missionTime &&
+                {(_data select 1) != objNull} &&
+                {(_data select 1) != player}
+                ) then {
+                    killcam_unit_pos = _unitPos;
+                    killcam_killer = _data select 1;
+                    killcam_killer_pos = _killerPos;
+                }
+                else {
+                    //everything failed, we set value we will detect later
+                    killcam_killer_pos = [0,0,0];
+                    killcam_unit_pos = ASLtoAGL eyePos _unit;
+                    killcam_killer = objNull;
+                };
+            }
+            else {
+                killcam_killer_pos = [0,0,0];
+                killcam_unit_pos = ASLtoAGL eyePos _unit;
+                killcam_killer = objNull;
+            };
+        };
+    
+        if (eg_instant_death) then {
+            private _damage = 0.5;
+            if (!isNil "killcam_LastHitDamage") then {
+                _damage = killcam_LastHitDamage;
+            };
+            cutText ["\n", "BLACK", 0.1, true];
+            [QGVAR(death), 0, true] call ace_common_fnc_setHearingCapability;
+            0 fadeSound 0;
+            [{
+                ["<t color='#FF0000'>YOU ARE DEAD</t>", 0, 0.4, 2, 0.5, 0, 1000] spawn BIS_fnc_dynamicText;
+            }, [], 1] call CBA_fnc_waitAndExecute;
+        } else {
+            private _damage = 0.5;
+            if (!isNil "killcam_LastHitDamage") then {
+                _damage = killcam_LastHitDamage;
+            };
+            [{
+                params ["_damage"];
+                cutText ["\n", "BLACK", (1.01 - _damage), true];
+                [QGVAR(death), 0, true] call ace_common_fnc_setHearingCapability;
+                [{
+                    params ["_damage"];
+                    0 fadeSound 0;
+                    [{
+                        ["<t color='#FF0000'>YOU ARE DEAD</t>", 0, 0.4, 2, 0.5, 0, 1000] spawn BIS_fnc_dynamicText;
+                    }, [], (1.01 - _damage)] call CBA_fnc_waitAndExecute;
+                }, [_damage], (1.01 - _damage)] call CBA_fnc_waitAndExecute;
+            }, [_damage]] call CBA_fnc_execNextFrame;
+        };
+    }];
 	
 	if (GVAR(StartOnSafe)) then {
 		[{(!isNull ace_player)}, {
-			if (currentWeapon player != "") then {
+			if !(currentWeapon player isEqualTo "") then {
 				[player, currentWeapon player, currentMuzzle player] call ace_safemode_fnc_lockSafety;
                 if (GETMVAR(StartOnSafe_Lowered,true)) then {
                     player action ["WeaponOnBack", player];
                 };
                 if (GETMVAR(StartOnSafe_Unloaded,true)) then {
                     private _magazineClass = currentMagazine player;
-                    player setAmmo [currentWeapon player, 0];
-                    player addMagazines [_magazineClass, 1];
```

Comment:
I'm assuming this was just a redundant call?

Reply:
This was the old style before removing the magazine was possible